test/e2e/qat/qatplugin_dpdk.go

```diff
@@ -55,6 +59,41 @@ func init() {
 	ginkgo.Describe("QAT plugin in DPDK mode [Device:qat] [Mode:dpdk]", describeQatDpdkPlugin)
 }
 
+func setQatAutoReset(cfgVal string) error {
+	matches, err := filepath.Glob(autoResetPattern)
+	if err != nil {
+		return err
+	}
+
+	for _, filePath := range matches {
+		err = os.WriteFile(filePath, []byte(cfgVal), 0600)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func injectError() error {
+	matches, err := filepath.Glob(injectErrorPattern)
+	if err != nil {
+		return err
+	}
+
+	for _, filePath := range matches {
+		file, err := os.OpenFile(filePath, os.O_WRONLY|os.O_TRUNC, 0600)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		os.Stdout = file
+		fmt.Println("1")
```

Comment:
os.WriteFile ?

Reply:
Good point.! I tried with os.WriteFile before I tried this way.
Unfortunately, that did not work.
I have no idea, but `inject_error` cannot be written in that way.

Reply:
WriteFile's content:
```
func WriteFile(name string, data []byte, perm FileMode) error {
	f, err := OpenFile(name, O_WRONLY|O_CREATE|O_TRUNC, perm)
	if err != nil {
		return err
	}
	_, err = f.Write(data)
	if err1 := f.Close(); err1 != nil && err == nil {
		err = err1
	}
	return err
}
```
At quick glance the only difference is 'O_CREATE' but that shouldn't matter. Did you check the error?

Reply:
hmm, i have no idea why it happened to be honest..

But, actually this topic is meaningless at this point.!
Mikko pointed out that we need to run not directly in the go code, but by running a pod that does the same thing in the system where the resources are.

Reply:
Thank you for your help though! :)